osbs/tekton.py

```diff
@@ -556,6 +559,16 @@ def has_not_finished(self):
     def was_cancelled(self):
         return self.status_reason == 'PipelineRunCancelled'
 
+    def wait_for_finish(self):
+        for _ in range(WAIT_RETRY):
+            if self.has_not_finished():
+                logger.info("Waiting for pipeline run '%s' to finish, sleep for %ss",
+                            self.pipeline_run_name, WAIT_RETRY_SECS)
+                time.sleep(WAIT_RETRY_SECS)
+            else:
+                logger.info("Pipeline run '%s' finished", self.pipeline_run_name)
+                break
```

Comment:
Should we raise an exception if the loop exits normally (i.e. the pipeline does not finish within the timeout)?

Reply:
there isn't any benefit in that, we will then later anyway call build_has_succeeded and build_not_finished (and cancel)

this is really just for getting some time for pipeline run to finish

Reply:
Ok, as long as the purpose of the function is documented this seems fine